google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -302,6 +320,99 @@ String getServiceNameFromAppEngineWebXml(XmlFile appengineWebXml) {
   }
 
   @Override
+  public void generateAppYaml(FlexibleRuntime runtime, Module module) {
+    Properties templateProperties =
+        FileTemplateManager.getDefaultInstance().getDefaultProperties();
+    templateProperties.put("RUNTIME", runtime.configLabel());
+
+    ApplicationManager.getApplication().runWriteAction(() -> {
+      generateFromTemplate(
+          AppEngineTemplateGroupDescriptorFactory.APP_YAML_TEMPLATE,
+          "appengine",
+          "app.yaml",
+          templateProperties,
+          module);
+    });
+  }
+
+  @Override
+  public void generateDockerfile(AppEngineFlexibleDeploymentArtifactType type, Module module) {
+    if (type == AppEngineFlexibleDeploymentArtifactType.UNKNOWN) {
+      throw new RuntimeException("Cannot generate Dockerfile for unknown artifact type.");
+    }
+
+    ApplicationManager.getApplication().runWriteAction(() -> {
+      PsiElement element = generateFromTemplate(
+          type == AppEngineFlexibleDeploymentArtifactType.JAR
+              ? AppEngineTemplateGroupDescriptorFactory.DOCKERFILE_JAR_TEMPLATE
+              : AppEngineTemplateGroupDescriptorFactory.DOCKERFILE_WAR_TEMPLATE,
+          "docker",
+          "Dockerfile",
+          FileTemplateManager.getDefaultInstance().getDefaultProperties(),
+          module);
+
+      if (element != null) {
+        // Remove the .docker extension to satisfy the Docker convention. This extension was added
+        // since the templating mechanism requires an extension or else a default template type of
+        // "java" will be assumed.
+        RenamePsiElementProcessor.DEFAULT.renameElement(
+            element,
+            "Dockerfile" /*newName*/,
+            UsageInfo.EMPTY_ARRAY,
+            null /*listener*/);
+      }
+    });
+  }
+
+  @Nullable
+  private PsiElement generateFromTemplate(String templateName, String templateDirectoryName,
+      String fileName, Properties templateProperties, Module module) {
+    FileTemplate configTemplate = FileTemplateManager.getDefaultInstance().getInternalTemplate(
+        templateName);
+    VirtualFile virtualFile = ModuleRootManager.getInstance(module).getContentRoots()[0]
```

Comment:
the generation works with regards to multi module projects - this is because the generation itself is scoped to a specific module (that has the facet).

I did uncover another related bug - the inspection of runtime custom vs java isn't correctly scoped to a module and breaks for multimodule. Will file a separate bug and fix separately